core/src/main/java/com/sequenceiq/cloudbreak/repository/RdsConfigRepository.java

```diff
@@ -1,59 +1,55 @@
 package com.sequenceiq.cloudbreak.repository;
 
-import java.util.Optional;
+import static com.sequenceiq.cloudbreak.authorization.OrganizationPermissions.Action.READ;
+
 import java.util.Set;
 
 import javax.transaction.Transactional;
 
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 
+import com.sequenceiq.cloudbreak.aspect.DisableHasPermission;
+import com.sequenceiq.cloudbreak.aspect.organization.CheckPermissionsByOrganizationId;
+import com.sequenceiq.cloudbreak.aspect.organization.OrganizationResourceType;
+import com.sequenceiq.cloudbreak.authorization.OrganizationResource;
 import com.sequenceiq.cloudbreak.domain.RDSConfig;
-import com.sequenceiq.cloudbreak.aspect.HasPermission;
 import com.sequenceiq.cloudbreak.service.EntityType;
 
 @EntityType(entityClass = RDSConfig.class)
 @Transactional(Transactional.TxType.REQUIRED)
-@HasPermission
-public interface RdsConfigRepository extends BaseRepository<RDSConfig, Long> {
-
-    @Query("SELECT r FROM RDSConfig r LEFT JOIN FETCH r.clusters WHERE r.owner= :user AND r.status = 'USER_MANAGED'")
-    Set<RDSConfig> findForUser(@Param("user") String user);
-
-    @Query("SELECT r FROM RDSConfig r LEFT JOIN FETCH r.clusters WHERE ((r.account= :account AND r.publicInAccount= true) OR r.owner= :user) "
-            + "AND r.status = 'USER_MANAGED' ")
-    Set<RDSConfig> findPublicInAccountForUser(@Param("user") String user, @Param("account") String account);
-
-    @Query("SELECT r FROM RDSConfig r LEFT JOIN FETCH r.clusters WHERE r.account= :account AND r.status = 'USER_MANAGED'")
-    Set<RDSConfig> findAllBasedOnAccount(@Param("account") String account);
+@DisableHasPermission
+@OrganizationResourceType(resource = OrganizationResource.RDS)
+public interface RdsConfigRepository extends OrganizationResourceRepository<RDSConfig, Long> {
 
-    @Query("SELECT r FROM RDSConfig r LEFT JOIN FETCH r.clusters WHERE r.owner= :owner and r.name= :name AND r.status = 'USER_MANAGED'")
-    RDSConfig findByNameInUser(@Param("name") String name, @Param("owner") String owner);
+    @CheckPermissionsByOrganizationId(action = READ)
+    @Query("SELECT r FROM RDSConfig r LEFT JOIN FETCH r.clusters WHERE r.organization.id = :orgId")
+    Set<RDSConfig> findAll(@Param("orgId") Long orgId);
 
-    @Query("SELECT r FROM RDSConfig r LEFT JOIN FETCH r.clusters WHERE r.name= :name and r.account= :account AND r.status = 'USER_MANAGED'")
-    RDSConfig findOneByName(@Param("name") String name, @Param("account") String account);
+    @CheckPermissionsByOrganizationId(action = READ, organizationIdIndex = 1)
+    @Query("SELECT r FROM RDSConfig r LEFT JOIN FETCH r.clusters WHERE r.name= :name AND r.organization.id = :orgId AND r.status = 'USER_MANAGED'")
+    RDSConfig findByName(@Param("name") String name, @Param("orgId") Long orgId);
```

Comment:
orgId is unnecessary here. especially since it is misused and default org id is passed in too many places. then permission can be checked with ```@CheckPermissionsByReturnValue```